oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -805,7 +805,10 @@ private long getNumPreviousDocuments() {
 
         private void addPreviousDocuments(Iterator<String> ids) throws IOException {
             while (ids.hasNext()) {
-                prevDocIdsToDelete.add(ids.next());
+                String id = ids.next();
+                if (id != null) {
+                    prevDocIdsToDelete.add(id);
+                }
```

Comment:
Was thinking to add a log.debug in case it is null. In which case you'd also need something meaningful to log - so maybe the `doc` or just the `id` (eg as `deletedId`) could be passed to this method and that could be logged. (and just at debug shouldn't hurt)
```suggestion
                } else if (log.isDebugEnabled()) {
                    log.debug("addPreviousDocuments: null id found via deletedId={}", deletedId);
                }
```

Reply:
I like this suggestion and will update the PR.